rust/main/agents/relayer/src/relayer/origin.rs

```diff
@@ -137,22 +137,19 @@ impl Factory for OriginFactory {
             res
         };
```

Comment:
Each IGP indexer runs as a separate tokio task. They all write to the same RocksDB instance, which already aggregates payments by message_id. This means a user can pay via any IGP and the relayer will see the total payment.